src/models/generation/econ_common.jl

```diff
@@ -44,3 +55,23 @@ EconThermal(;   capacity = 0.0,
             shutdncost = 0.0,
             annualcapacityfactor = nothing
         ) = EconThermal(capacity, variablecost, fixedcost, startupcost, shutdncost, annualcapacityfactor)
+
+"""Enables deserialization of EconThermal. The default implementation can't figure out the
+variablecost Union.
+"""
+function JSON2.read(io::IO, ::Type{T}) where {T <: EconThermal}
+    data = JSON2.read(io)
+    if data.variablecost isa Array
+        variablecost = Vector{Tuple{Float64, Float64}}()
```

Comment:
I have a preference for using `Tuple{Float64, Float64}[]` instead of `Vector{Tuple{Float64, Float64}}()`.

Reply:
Left as-is per our discussion.